ext/reflection/php_reflection.stub.php

```diff
@@ -829,3 +829,24 @@ public function getCallable(): callable {}
 
     public function getTrace(int $options = DEBUG_BACKTRACE_PROVIDE_OBJECT): array {}
 }
+
+/**
+ * @strict-properties
+ * @not-serializable
+ */
+final class ReflectionConstant
+{
+    public string $name;
+
+    public function __construct(string $name) {}
+
+    public function getName(): string {}
+
+    public function getNamespaceName(): string {}
+
+    public function getShortName(): string {}
+
+    public function getValue(): mixed {}
+
+    public function isDeprecated(): bool {}
```

Comment:
slightly related: theoretically, global constants could also store a doc comment if we wanted to, just like all the other symbols. do we want to add support for this? (meaning: I can work on it)

Reply:
Should be possible, but it's out of scope for this RFC. Unlike classes, functions and other declarations, `zend_constant`s are created at runtime. Thus, there would need to be a way to pass the doc comment to the declaration. Both op1 and op2 are currently occupied for `ZEND_DECLARE_CONST`.

Reply:
Ah no, I didn't mean to work on this in the scope of this PR :) I was mainly wondering if you think it's a good idea to add support for the feature. But I'm not attached to it too much, and I definitely don't want to change the opcode/add new opcode for such a feature. To be honest, I forgotten about userland constants, and only meant to change `zend_register_constant()`... But only dealing with internal constants would be pointless, that's true.

(Regarding the taken 2 operands of the `ZEND_DECLARE_CONST` opcode: I was aware of this restriction because I was trying to add support for type declaration of global constants.. Finally, I changed my mind due to the above fact.)